core/spec/models/spree/order_promotion_spec.rb

```diff
@@ -0,0 +1,54 @@
+require 'spec_helper'
+
+describe Spree::OrderPromotion, type: :model do
+  subject { create(:order_promotion, order: order, promotion: promotion) }
+
+  let(:order) { create(:order_with_line_items) }
+  let(:promotion) { create(:promotion_with_item_adjustment, code: 'test') }
+
+  shared_context 'apply promo' do
+    before do
+      order.coupon_code = promotion.code
+      Spree::PromotionHandler::Coupon.new(order).apply
+      order.save!
+      order.all_adjustments.promotion.update_all(amount: -5.0)
```

Comment:
Maybe we could use `adjustment_rate` transient field of `with_line_item_adjustment` factory to set this value (https://github.com/spree/spree/blob/master/core/lib/spree/testing_support/factories/promotion_factory.rb#L16)?

Reply:
@piotrleniec-spark we want to test it explicitly here - this avoids any side-effect and false positives